Source/Peripheral.swift

```diff
@@ -127,29 +127,35 @@ public class Peripheral {
 	 Immediately after that `.Complete` is emitted.
 	 */
 	public func discoverServices(serviceUUIDs: [CBUUID]?) -> Observable<[Service]> {
-		let observable = peripheral.rx_didDiscoverServices
+        if let identifiers = serviceUUIDs, let services = self.services?.filter({ identifiers.contains($0.UUID) })
+                                           where identifiers.count == services.count {
+            return ensureValidPeripheralState(Observable.just(services))
+        } else {
```

Comment:
else branch is not necessary because of early return. We will reduce indentation level and gain more readability